core/trino-main/src/main/java/io/trino/execution/querystats/QueryStatsCollectorFactory.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.execution.querystats;
+
+import io.trino.execution.QueryManagerConfig;
+
+import javax.inject.Inject;
+
+public class QueryStatsCollectorFactory
```

Comment:
Do we need this factory. IMO we could just create `DefaultQueryStatsCollector` in code. You already have `QueryManagerConfig` in `LocalDispatchQueryFactory`

Reply:
+1 looks unnecessary